c++/src/kj/BUILD

```diff
@@ -0,0 +1,218 @@
+load("//c++/src/kj:configure.bzl", "kj_configure")
+
+kj_configure()
+
+kj_sources = [
+    "arena.c++",
+    "array.c++",
+    "common.c++",
+    "debug.c++",
+    "encoding.c++",
+    "exception.c++",
+    "filesystem.c++",
+    "filesystem-disk-unix.c++",
+    "hash.c++",
+    "io.c++",
+    "list.c++",
+    "main.c++",
+    "memory.c++",
+    "mutex.c++",
+    "parse/char.c++",
+    "refcount.c++",
+    "source-location.c++",
+    "string-tree.c++",
+    "string.c++",
+    "table.c++",
+    "test-helpers.c++",
+    "thread.c++",
+    "time.c++",
+    "units.c++",
+]
+
+kj_headers = [
+    "arena.h",
+    "array.h",
+    "common.h",
+    "debug.h",
+    "encoding.h",
+    "exception.h",
+    "filesystem.h",
+    "function.h",
+    "hash.h",
+    "io.h",
+    "list.h",
+    "main.h",
+    "map.h",
+    "memory.h",
+    "miniposix.h",
+    "mutex.h",
+    "one-of.h",
+    "parse/char.h",
+    "parse/common.h",
+    "refcount.h",
+    "source-location.h",
+    "std/iostream.h",
+    "string-tree.h",
+    "string.h",
+    "table.h",
+    "test.h",
+    "thread.h",
+    "threadlocal.h",
+    "time.h",
+    "tuple.h",
+    "units.h",
+    "vector.h",
+    "win32-api-version.h",
+    "windows-sanity.h",
+]
+
+cc_library(
+    name = "kj",
+    srcs = kj_sources,
+    hdrs = kj_headers,
+    include_prefix = "kj",
+    linkopts = ["-lpthread"] + select({
+        ":has_libdl": ["-ldl"],
+        "//conditions:default": [],
+    }),
+    visibility = ["//visibility:public"],
+    deps = [":kj-defines"],
+)
+
+kj_async_sources = [
+    "async-io-internal.h",
+    "async-io-unix.c++",
+    "async-io-win32.c++",
+    "async-io.c++",
+    "async-unix.c++",
+    "async-win32.c++",
+    "async.c++",
+    "timer.c++",
+]
+
+kj_async_headers = [
+    "async-inl.h",
+    "async-io.h",
+    "async-prelude.h",
+    "async-queue.h",
+    "async-unix.h",
+    "async-win32.h",
+    "async.h",
+    "timer.h",
+]
+
+cc_library(
+    name = "kj-async",
+    srcs = kj_async_sources,
```

Comment:
Can we make the list of sources explicit attributes as opposed to variables? It's more idiomatic and the variables aren't being used anywhere else.